src/world/Field.cpp

```diff
@@ -72,6 +74,16 @@ void Field::initFieldVectors() {
     Vector2 rpl_begin = rightPenaltyLine.value().begin;
     Vector2 rpl_end = rightPenaltyLine.value().end;
     rightPenaltyPoint = rpl_begin + ((rpl_end - rpl_begin) * 0.5);
+
+    leftPenaltyLineBottom = leftPenaltyLine->begin;
+    leftPenaltyLineTop = leftPenaltyLine->end;
+    rightPenaltyLineBottom = rightPenaltyLine->begin;
+    rightPenaltyLineTop = rightPenaltyLine->end;
+
+    bottomLeftCorner = Vector2(leftmostX.value(), bottommostY.value());
+    topLeftCorner = Vector2(leftmostX.value(), topmostY.value());
+    bottomRightCorner = Vector2(rightmostX.value(), bottommostY.value());
+    topRightCorner = Vector2(rightmostX.value(), topmostY.value());
```

Comment:
This needs 100% tests. + There are no safety checks, ssl data can be ordered wrong! This will cause problems down the line.

Reply:
> I agree with Lukas that if you change classes like world, field we should generally aim to fix the tests. Right now, literally your entire PR rests on some assumptions which I _know_ are not satisfied at all times by the SSL data. I did not look through the entire PR but it seems it's mostly very minor style changes in a lot of files again.
> 
> Documentation wise: I also think it's too much. Generally comments should be there to explain 'why' something works or why you can do it, e.g. provide further insight. The naming should make clear WHAT the object or parameter in question is in 99% of the cases already. Even a random programmer from the street who has learned roughly what SSL does will understand what Vector2 TopRightCorner represents.
> 
> Testing is highly important. If you take this approach, I'm not sure it's wise if you touch the DefencePositionCoach, because you 100% need to make better unit tests than the ones already existing now to ensure behavior does not change. The code there is quite complicated and slightly hacky at times.

I guess if I ask a random software developer on the streets whether the topRightCorner has positive/negative x/y-coordinates then he would not be able to tell it or guess it correctly with 25%. Regarding the tests I agree with you. I did not saw that Travis rejected it, so I will indeed fix the test cases before merging.

Reply:
I am not even talking about the fact tests are breaking. I am talking that about the fact you did not write any new ones.